core/templates/dev/head/pages/exploration_player/AudioBarDirective.js

```diff
@@ -28,19 +28,108 @@ oppia.directive('audioControls', [
       },
       templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
         '/pages/exploration_player/' +
-        'audio_controls_directive.html'),
+        'audio_bar_directive.html'),
       controller: [
-        '$scope', 'AudioTranslationManagerService', 'AudioPlayerService',
-        'LanguageUtilService', 'AssetsBackendApiService',
+        '$scope', '$interval', 'AudioTranslationManagerService',
+        'AudioPlayerService', 'LanguageUtilService', 'AssetsBackendApiService',
         'AutogeneratedAudioPlayerService', 'PlayerPositionService',
+        'WindowDimensionsService', 'EVENT_ACTIVE_CARD_CHANGED',
         function(
-            $scope, AudioTranslationManagerService, AudioPlayerService,
-            LanguageUtilService, AssetsBackendApiService,
-            AutogeneratedAudioPlayerService, PlayerPositionService) {
+            $scope, $interval, AudioTranslationManagerService,
+            AudioPlayerService, LanguageUtilService, AssetsBackendApiService,
+            AutogeneratedAudioPlayerService, PlayerPositionService,
+            WindowDimensionsService, EVENT_ACTIVE_CARD_CHANGED) {
           // This ID is passed in to AudioPlayerService as a means of
           // distinguishing which audio directive is currently playing audio.
           var directiveId = Math.random().toString(36).substr(2, 10);
 
+          $scope.isAudioBarExpanded = false;
+          $scope.hasPressedPlayButtonOnce = false;
+
+          $scope.languagesInExploration =
+            AudioTranslationManagerService.getLanguageOptionsForDropdown();
+          $scope.selectedLanguage = {
+            value: AudioTranslationManagerService.getCurrentAudioLanguageCode()
+          };
+
+          $scope.$on(EVENT_ACTIVE_CARD_CHANGED, function() {
+            if ($scope.isAudioBarExpanded) {
+              $scope.onSpeakerIconClicked();
+            }
+          });
+
+          $scope.getProgressBarMode = function() {
+            if ($scope.audioLoadingIndicatorIsShown) {
+              return 'indeterminate';
+            } else {
+              return 'determinate';
+            }
+          };
+
+          $scope.languageSelected = function() {
```

Comment:
$scope.onLanguageSelected (use "on..." for event handlers)

Does this apply only when the underlying value is changed? If so, let's go with $scope.onNewLanguageSelected instead for clarity.

Reply:
changed to onNewLanguagesSelected